### PR TITLE
fix: Fixed handlepop function  on close tab for environment [#123]

### PR DESCRIPTION
**Pull request by Abhishek9503**
## Description

In this  PR i have added a condition which is  triggered  when i close the environement  tab without saving the tab , so now a popup will appear for confirmation .

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

**Comment by pratikshinde590** [comment]
Please attach the issue link

**Comment by pratikshinde590** [comment]
Please attach the issue link

File: src/packages/@app/pages/EnvironmentPage/EnvironmentPage.ViewModel.ts [hunk 1]
```diff
@@ -319,4 +319,77 @@ export class EnvironmentViewModel {
...
+
+    /**
+   * @description - saves environment to the mongo server
+   */
+    public saveEnvironment = async (_tab) => {
+      const currentEnvironment = _tab;
+      const activeWorkspace = await this.workspaceRepository.readWorkspace(
+        currentEnvironment.workspaceId,
+      );
+      const guestUser = await this.guestUserRepository.findOne({
```
Comment (itsmdasifraza) [inline src/packages/@app/pages/EnvironmentPage/EnvironmentPage.ViewModel.ts:331]: is `workspaceId` exist on `currentEnvironment`

**Review by itsmdasifraza — APPROVED** [review]

**Pull request merged**